modules/core/src/test/scala/ch/epfl/bluebrain/nexus/rdf/PathSpec.scala

```diff
@@ -92,5 +92,30 @@ class PathSpec extends WordSpecLike with Matchers with Inspectors with EitherVal
     "eq" in {
       Eq.eqv(abcd.right.value, Segment("d", Path("/a/b//c/").right.value)) shouldEqual true
     }
+
+    "start with slash" in {
+      val cases = List("/", "///", "/a/b/c/d", "/a/b/c/d/")
+      forAll(cases) {
+        case (str) => Path(str).right.value.startWithSlash shouldEqual true
+      }
+    }
+
+    "does not start with slash" in {
+      val cases = List(Empty, Segment("d", Slash(Segment("c", Slash(Slash(Segment("b", Slash(Segment("a", Empty)))))))))
+      forAll(cases) {
+        case (p) => p.startWithSlash shouldEqual false
+      }
+    }
+
+    "reverse" in {
```

Comment:
add tests:
*   when path is empty
*   when path ends with a slash
*   when path is just a slash